sources/sql_database/__init__.py

```diff
@@ -30,11 +35,12 @@ def sql_database(
     table_names: Optional[List[str]] = dlt.config.value,
     chunk_size: int = 50000,
     backend: TableBackend = "sqlalchemy",
-    detect_precision_hints: Optional[bool] = dlt.config.value,
```

Comment:
now when I think of it.. let's keep this option and mark as deprecated. when true it should set the highest reflection level. we'll have a lot of users complaining that production code stopped working...